tests/component/test_stream_readers.py

```diff
@@ -545,3 +547,591 @@ def test___power_binary_reader___read_many_sample_with_wrong_dtype___raises_erro
+) -> nidaqmx.Task:
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_lines[:4].name),
+        line_grouping=LineGrouping.CHAN_PER_LINE,
+    )
+    return task
+
+
+@pytest.fixture
+def di_single_channel_port_byte_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 port 1/2 have 8 lines each
+    task.di_channels.add_di_chan(
+        sim_6363_device.di_ports[1].name,
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    return task
+
+
+@pytest.fixture
+def di_multi_channel_port_byte_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 port 1/2 have 8 lines each, but we need to skip port 0, which has 32 lines
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_lines[32:40].name),
+        line_grouping=LineGrouping.CHAN_PER_LINE,
+    )
+    return task
+
+
+@pytest.fixture
+def di_single_channel_port_uint16_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 port 1/2 have 8 lines each, combine to 16
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_ports[1:].name),
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    return task
+
+
+@pytest.fixture
+def di_multi_channel_port_uint16_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 doesn't have a 16-line port. We could merge two 8-line ports together, but the data
+    # returned is based on the size of the port. For example, line 9 in the task would be line 0 of
+    # the second 8-bit port. Instead of the data being 0x00F0, it would be 0xF000 (little-endian).
+    # Anyways, its harder to reason about, so we'll only use 8 lines. DAQ will happily extend the
+    # data to the larger type.
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_lines[32:40].name),
+        line_grouping=LineGrouping.CHAN_PER_LINE,
+    )
+    return task
+
+
+@pytest.fixture
+def di_single_channel_port_uint32_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 port 0 has 32 lines
+    task.di_channels.add_di_chan(
+        sim_6363_device.di_ports[0].name,
+        line_grouping=LineGrouping.CHAN_FOR_ALL_LINES,
+    )
+    return task
+
+
+@pytest.fixture
+def di_multi_channel_port_uint32_task(
+    task: nidaqmx.Task, sim_6363_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    # 6363 port 0 has 32 lines
+    task.di_channels.add_di_chan(
+        flatten_channel_string(sim_6363_device.di_lines[:32].name),
+        line_grouping=LineGrouping.CHAN_PER_LINE,
+    )
+    return task
+
+
+def _get_expected_digital_data_for_sample(num_lines: int, sample_number: int) -> int:
+    # Simulated digital signals "count" from 0 in binary within each group of 8 lines.
```

Comment:
Reading 10 samples counts from 0b00000000 to 0b00001010, so it only exercises the lower 4 bits.